### PR TITLE
Add job that passed when all other CI jobs have completed

### PR DESCRIPTION
**Pull request by peternied**
### Description
GitHub allows 'requiring' a job to be run before a pull request can be merged, by adding a new job that 'needs' all the other ci jobs, this job will run last and only if they all pass.

After this is merged someone from the admin team can update the repo settings to enforce this requirement.

### Check List
- [ ] ~New functionality includes testing~
  - [ ] ~All tests pass, including unit test, integration test and doctest~
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by AndreKurait — APPROVED** [review]

**Comment by peternied** [comment]
@opensearch-project/admin Can you please add a branch protection on main that required check for `CI / 
all-ci-checks-pass`?

**Comment by gaiksaya** [comment]
> @opensearch-project/admin Can you please add a branch protection on main that required check for `CI / all-ci-checks-pass`?

image
Done! Also added `require code reviews from code owners` Hope that's okay!

**Comment by peternied** [comment]
@gaiksaya Thanks for the fast turn around!

**Pull request merged**
